core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/DefaultNamingScheme.java

```diff
@@ -22,9 +24,17 @@
   private NameValidator sampleAliasValidator = new DefaultSampleAliasValidator();
   private NameGenerator<Sample> sampleAliasGenerator = null;
   private NameValidator libraryAliasValidator = new DefaultLibraryAliasValidator();
-  private NameGenerator<Library> libraryAliasGenerator = new DefaultLibraryAliasGenerator();
+  private DefaultLibraryAliasGenerator libraryAliasGenerator = new DefaultLibraryAliasGenerator();
```

Comment:
`NameGenerator` must remain the declared type - configuration may cause this to be replaced by a different generator